cstrml/staking/src/lib.rs

```diff
@@ -1946,6 +1948,24 @@ impl<T: Trait> Module<T> {
     }
 }
 
+pub trait EnhancedBalanceInterface<AccountId, Balance>: system::Trait {
+    fn free_balance_for_staking_only(who: &AccountId) -> Balance;
+}
+
+impl<T: Trait> EnhancedBalanceInterface<<T as system::Trait>::AccountId, BalanceOf<T>> for T where T: balances::Trait
+{
+    fn free_balance_for_staking_only(who: &<T as system::Trait>::AccountId) -> BalanceOf<T>
+    {
+        let to_balanceof = |e: T::Balance| <T::CurrencyToVote as Convert<u128, BalanceOf<T>>>::convert(TryInto::<u128>::try_into(e).ok().unwrap());
+        let sum = <balances::Module<T>>::locks(who).iter().fold(Zero::zero(), |acc: <T as balances::Trait>::Balance, value| {
```

Comment:
There are two concerns for such design. 
- `locks` is not in any trait and cannot be used directly. I have to include balance as one dependency and extend its ability.
- There are two types `T::Balance` and `BalanceOf<T>`. I have to use u128 to covert one to another.

Reply:
Another option is to include `Balance` module and modify it directly.

Reply:
As for `Maxwell 1.0`, we take ***solution1***, import balance should be a long-term solution. And `locks` should be and only be used in `staking` and `market` pallets inside crust repo.

Reply:
Discussed offline, choose `solution2` since we need balance in the future anyway.

Reply:
> As for `Maxwell 1.0`, we take ***solution1***, import balance should be a long-term solution. And `locks` should be and only be used in `staking` and `market` pallets inside crust repo.

Talked offline, considering the long term repeated work, we take sol2 as our final offer.